library/snapshot.py

```diff
@@ -2033,438 +2239,217 @@ def get_json_from_args(args):
-        "-lv",
-        "--logicalvolume",
-        nargs="?",
-        action="store",
-        default=None,
-        dest="logical_volume",
-        help="logical volume to snapshot",
-    )
-
-    # arguments for operations that do verify
-    verify_parser = argparse.ArgumentParser(add_help=False)
-    verify_parser.add_argument(
-        "-v",
-        "--verify",
-        action="store_true",
-        default=False,
-        dest="verify",
-        help="verify VGs and LVs have snapshots",
-    )
-
-    # arguments for operations that deal with required space
-    req_space_parser = argparse.ArgumentParser(add_help=False)
-    # TODO: range check required space - setting choices to a range makes the help ugly.
-    req_space_parser.add_argument(
-        "-r",
-        "--requiredspace",
-        dest="required_space",
-        required=False,
-        type=int,  # choices=range(10,100)
-        default=20,
-        help="percent of required space in the volume group to be reserved for snapshot",
-    )
-
-    # arguments for operations that deal with mount of filesytems
-    mountpoint_parser = argparse.ArgumentParser(add_help=False)
-    mountpoint_parser.add_argument(
-        "-m",
-        "--mountpoint",
-        dest="mountpoint",
-        required=False,
-        type=str,
-        help="mount point for block device",
-    )
-
-    # arguments for operations that deal with mount of filesytems
-    mount_parser = argparse.ArgumentParser(add_help=False)
-    mount_parser.add_argument(
-        "-b",
-        "--blockdev",
-        dest="blockdev",
-        required=False,
-        type=str,
-        help="mount point for block device",
-    )
-    mount_parser.add_argument(
-        "-t",
-        "--type",
-        dest="fstype",
-        required=False,
-        default="",
-        type=str,
-        help="filesystem type",
-    )
-    mount_parser.add_argument(
-        "-o",
-        "--options",
-        dest="options",
-        required=False,
-        type=str,
-        help="mount options",
+def run_module():
+    logger.info("run_module()")
+    vg_include = None
+    # define available arguments/parameters a user can pass to the module
+    # available arguments/parameters that a user can pass
+    module_args = dict(
+        snapshot_lvm_action=dict(type="str"),
+        ansible_check_mode=dict(type=bool),
+        snapshot_lvm_all_vgs=dict(type=bool),
+        snapshot_lvm_verify_only=dict(type=bool),
+        snapshot_lvm_mount_origin=dict(type=bool),
+        snapshot_lvm_mountpoint_create=dict(type=bool),
+        snapshot_lvm_unmount_all=dict(type=bool),
+        snapshot_lvm_percent_space_required=dict(type="str"),
+        snapshot_lvm_vg=dict(type="str"),
+        snapshot_lvm_lv=dict(type="str"),
+        snapshot_lvm_snapset_name=dict(type="str"),
+        snapshot_lvm_mount_options=dict(type="str"),
+        snapshot_lvm_mountpoint=dict(type="str"),
+        snapshot_lvm_set=dict(type="str"),
```

Comment:
This should be a `list` of `dict` - see the storage role for examples e.g. https://github.com/linux-system-roles/storage/blob/main/library/blivet.py#L2131
```yaml
        snapshot_lvm_set=dict(type="list", elements="dict",
                                                  options=dict(name=dict(type="str"),
                                                                         vg=dict(type="str"),
                                                                         lv=dict(type="str"),
                                                                         percent_space_required=dict(type="int"),
....
```

Reply:
Sorry - not a list of dict but just a `dict`